### ISSUE TITLE
Only start sending messages when connected to > 1 peers?

### ISSUE THREAD

**Issue by kdeme** [status-im/nim-eth#106]
Perhaps we should add a check of x connected peers before we start sending Whisper messages at the specified interval.

If a node is only connected to 1 peer, and that peer is malicious, it could decide not to forward the node its messages.
In this case, the malicious peer could monitor with all it's connected peers if any duplicates of the node its messages are arriving. If this does not occur, it could mean that the messages originates from that node. It could of course also just mean, that the TTL is not long enough to get forwarded all the way to this peer. However, with a low amount of nodes (which Status kinda has), and over several messages, the malicious peer could probably deduce this. 

This might be a none issue, especially if a node is only for a very limited time (like @ startup) connected to only 1 node.

**Comment by arnetheduck** [comment]
sounds like something that could be generalized into a privacy score/metric where peer diversity and other heuristics are considered. of course, it's gameable, but at least then then we can add other components than peer count specifically to make it harder.

**Comment by cheatfate** [comment]
It is better to consider only `outgoing` peers, not `incoming` peers.

**Comment by kdeme** [comment]
Whisper/Waku code was moved out of nim-eth
